### PR TITLE
Add git credential fetch support, minor bug fixes

### PR DESCRIPTION
**Pull request by kotwanikunal**
Signed-off-by: Kunal Kotwani <kkotwani@amazon.com>

### Description
- Updated the github token fetch mechanism to use the github bot fetching mechanism
- Minor Fix: Added `stackNameSuffix` to prevent two builds with the same build id use the same infra stack
- Minor Fix: Fixed typo for `config` under the perf test script

### Issues Resolved
- #129 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by ankitkala — APPROVED** [review]
LGTM

File: vars/runPerfTestScript.groovy [hunk 1]
```diff
@@ -4,24 +4,28 @@ void call(Map args = [:]) {
...
-        "--config config.yml",
-        args.insecure ? "--without-security" : "",
-        isNullOrEmpty(args.workload) ? "" : "--workload ${args.workload}",
-        isNullOrEmpty(args.testIterations) ? "" : "--test-iters ${args.testIterations}",
-        isNullOrEmpty(args.warmupIterations) ? "" : "--warmup-iters ${args.warmupIterations}",
-        isNullOrEmpty(args.component) ? "" : "--component ${args.component}"
-    ].join(' '))
+    String stackNameSuffix = isNullOrEmpty(args.stackNameSuffix) ? 'perf-test' : args.stackNameSuffix
+
+    withCredentials([usernamePassword(credentialsId: "${GITHUB_BOT_TOKEN_NAME}", usernameVariable: 'GITHUB_USER', passwordVariable: 'GITHUB_TOKEN')]) {
```
Comment (peterzhuamazon) [inline vars/runPerfTestScript.groovy:14]: I dont see anywhere define the actual token id, are you setting it in the jenkins global env?

Comment (kotwanikunal) [reply vars/runPerfTestScript.groovy:14]: It is defined as a Jenkins global env var. More usages - https://github.com/opensearch-project/opensearch-build/blob/c125b697815ca9887873752e0c3c7bdda0dfe00d/vars/createReleaseTag.groovy#L12, https://github.com/opensearch-project/opensearch-build/blob/de56d0fef90962339e4b292890c69c5948a3663a/vars/signArtifacts.groovy#L28

**Review by peterzhuamazon — APPROVED** [review]

**Comment by kotwanikunal** [comment]
@opensearch-project/engineering-effectiveness I need some help merging this in. I do not currently have write permissions to this repo.

**Pull request merged**